service/src/main/java/io/camunda/service/search/filter/FlownodeInstanceFilter.java

```diff
@@ -7,15 +7,151 @@
  */
 package io.camunda.service.search.filter;
 
+import static io.camunda.util.CollectionUtil.addValuesToList;
+import static io.camunda.util.CollectionUtil.collectValuesAsList;
+
 import io.camunda.util.ObjectBuilder;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
 
-public record FlownodeInstanceFilter() implements FilterBase {
+public record FlownodeInstanceFilter(
+    List<Long> keys,
```

Comment:
I wonder if we should use `flowNodeKeys` to be consistent with the other filters
We have used `decisionRequirementsKeys` for instance to be consistent with Zeebe API namings, but it seems that FlowNodeInstance is not part of the Zeebe API

Reply:
Good point, it would then `flowNodeInstanceKeys` etc.